### ISSUE TITLE
Is there a solution similar to Evt differ in QPnano?

### ISSUE THREAD

**Issue by Femtometer** [QuantumLeaps/qpn#3]
I'm using QPn on serial port. 
When a baudrate change CMD is received, I need to response before hardware re-init.
But NewBaudEvt is POST along with SerialNeedSendEvt.
NewBaudEvt makes serial stop when sending is need.

**Comment by quantum-leaps** [comment]
I'm not sure that I fully understand the problem, but presuming that you have a dedicated active object (AO) that exclusively accesses the serial port, then upon the reception of the NewBaudEvt, you can start sending the response and transition to the special "busy" state. Only once the sending completes (which must somehow generate some "SendingDone" event), you can actually command the serial port to change the baud rate. You can also use the ["Deferred Event"](https://www.state-machine.com/doc/Pattern_DeferredEvent.pdf) state pattern to defer all other events while in the "busy" state.
